wagtail/wagtailcore/query.py

```diff
@@ -394,8 +415,15 @@ class SpecificIterable(BaseIterable):
         def __iter__(self):
             return specific_iterator(self.queryset)
 
+    class DeferredSpecificIterable(BaseIterable):
+        def __iter__(self):
+            return specific_iterator(self.queryset, defer=True)
+
 else:
     from django.db.models.query import QuerySet
 
     class SpecificQuerySet(QuerySet):
         iterator = specific_iterator
+
+    class DeferredSpecificQuerySet(QuerySet):
+        iterator = functools.partial(specific_iterator, defer=True)
```

Comment:
Ah, with `functools.partial` it should work. Ok, then I do not understand the reason of test failures...